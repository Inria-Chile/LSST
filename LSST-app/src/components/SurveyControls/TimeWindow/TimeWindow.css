.time-window-wrapper {
    display: inline-block;    
    cursor: pointer;
}

.time-window-selection-span {
    font-weight: bold;
    width: 70px;
    display: inline-block;
    text-align: center;
    cursor: pointer;
}

.time-window-selection-input:checked + span {
    border-bottom: 6px solid #fff;
    padding-bottom: 0px;
    color: #fff;
    cursor: pointer;
}

.time-window-selection-span {
    padding-bottom: 2px;
    border-bottom: 2px solid #96a3b3;
    cursor: pointer;
}

input.time-window-selection-input {    
    display: none;
}

.time-window-selection-label {
    height: 40px;
    cursor: pointer;
    margin-bottom: 0px;
    border-radius: 5px;
}