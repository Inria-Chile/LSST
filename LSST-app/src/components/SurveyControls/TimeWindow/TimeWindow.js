import React, { Component } from 'react';
import './TimeWindow.css';

class TimeWindow extends Component {
    static timeWindowOptions = {
        "10s": 10,
        "1min": 60,
        "10min": 600,
        "1d": 24*60*60,
        "10d": 240*60*60,
        "1m": 720*60*60,
        "6m": 4320*60*60,
        "1y": 8640*60*60,
        "Max": Infinity,

    }
    constructor(props) {
        super(props);
        this.state = {
            timeWindow: TimeWindow.timeWindowOptions[Object.keys(TimeWindow.timeWindowOptions)[0]],
        }
    }

    setDate = () => {
        this.props.setDataByDate(this.state.startDate, this.state.endDate);
    }

    handleChange = (changeEvent) => {
        let tw = parseInt(changeEvent.target.value, 10)
        this.props.setTimeWindow(tw);
        this.setState({
            timeWindow: tw,
        })
    }

    render() {
        console.log(this.state.timeWindow)
        return (
            <div className="time-window">
                <div>
                    <h6>Time window</h6>
                </div>
                {
                    Object.keys(TimeWindow.timeWindowOptions).map((k, index) => {
                        return (
                            <label key={k} className="time-window-selection-label" htmlFor={k}>
                                <input className="time-window-selection-input" id={k} 
                                        checked={TimeWindow.timeWindowOptions[k]===this.state.timeWindow}
                                        type="radio" name="field" value={TimeWindow.timeWindowOptions[k]}
                                        onChange={this.handleChange}/>
                                <span className="time-window-selection-span">{k}</span>
                            </label>
                        );
                    })
                }
            </div>
        );
    }
}

export default TimeWindow;
