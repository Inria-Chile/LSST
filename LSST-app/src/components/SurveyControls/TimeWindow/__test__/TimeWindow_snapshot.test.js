import React from 'react';
import TimeWindow from '../TimeWindow';
import renderer from 'react-test-renderer';

/*---- Snapshot testing ----*/
describe('TimeWindow_Test',function(){

    it('renders correctly without shallowRender', () => {
      let setTimeWindow = jest.fn();
      const tree = renderer
          .create(<TimeWindow setTimeWindow={setTimeWindow} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
});
