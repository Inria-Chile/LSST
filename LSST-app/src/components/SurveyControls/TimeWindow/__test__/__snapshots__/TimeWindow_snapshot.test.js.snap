// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeWindow_Test renders correctly without shallowRender 1`] = `
<div
  className="time-window"
>
  <div>
    <h6>
      Time window
    </h6>
  </div>
  <label
    className="time-window-selection-label"
    htmlFor="10s"
  >
    <input
      checked={true}
      className="time-window-selection-input"
      id="10s"
      name="field"
      onChange={[Function]}
      type="radio"
      value={10}
    />
    <span
      className="time-window-selection-span"
    >
      10s
    </span>
  </label>
  <label
    className="time-window-selection-label"
    htmlFor="1min"
  >
    <input
      checked={false}
      className="time-window-selection-input"
      id="1min"
      name="field"
      onChange={[Function]}
      type="radio"
      value={60}
    />
    <span
      className="time-window-selection-span"
    >
      1min
    </span>
  </label>
  <label
    className="time-window-selection-label"
    htmlFor="10min"
  >
    <input
      checked={false}
      className="time-window-selection-input"
      id="10min"
      name="field"
      onChange={[Function]}
      type="radio"
      value={600}
    />
    <span
      className="time-window-selection-span"
    >
      10min
    </span>
  </label>
  <label
    className="time-window-selection-label"
    htmlFor="1d"
  >
    <input
      checked={false}
      className="time-window-selection-input"
      id="1d"
      name="field"
      onChange={[Function]}
      type="radio"
      value={86400}
    />
    <span
      className="time-window-selection-span"
    >
      1d
    </span>
  </label>
  <label
    className="time-window-selection-label"
    htmlFor="10d"
  >
    <input
      checked={false}
      className="time-window-selection-input"
      id="10d"
      name="field"
      onChange={[Function]}
      type="radio"
      value={864000}
    />
    <span
      className="time-window-selection-span"
    >
      10d
    </span>
  </label>
  <label
    className="time-window-selection-label"
    htmlFor="1m"
  >
    <input
      checked={false}
      className="time-window-selection-input"
      id="1m"
      name="field"
      onChange={[Function]}
      type="radio"
      value={2592000}
    />
    <span
      className="time-window-selection-span"
    >
      1m
    </span>
  </label>
  <label
    className="time-window-selection-label"
    htmlFor="6m"
  >
    <input
      checked={false}
      className="time-window-selection-input"
      id="6m"
      name="field"
      onChange={[Function]}
      type="radio"
      value={15552000}
    />
    <span
      className="time-window-selection-span"
    >
      6m
    </span>
  </label>
  <label
    className="time-window-selection-label"
    htmlFor="1y"
  >
    <input
      checked={false}
      className="time-window-selection-input"
      id="1y"
      name="field"
      onChange={[Function]}
      type="radio"
      value={31104000}
    />
    <span
      className="time-window-selection-span"
    >
      1y
    </span>
  </label>
  <label
    className="time-window-selection-label"
    htmlFor="Max"
  >
    <input
      checked={false}
      className="time-window-selection-input"
      id="Max"
      name="field"
      onChange={[Function]}
      type="radio"
      value={Infinity}
    />
    <span
      className="time-window-selection-span"
    >
      Max
    </span>
  </label>
</div>
`;
